op-batcher/batcher/espresso.go

```diff
@@ -897,21 +916,91 @@ func (l *BatchSubmitter) registerBatcher(ctx context.Context) error {
 
 	batchAuthenticator, err := bindings.NewBatchAuthenticator(l.RollupConfig.BatchAuthenticatorAddress, l.L1Client)
 	if err != nil {
-		return fmt.Errorf("failed to create batch authenticator contract bindings: %w", err)
+		return fmt.Errorf("failed to create BatchAuthenticator contract bindings: %w", err)
+	}
+
+	verifierAddress, err := batchAuthenticator.EspressoTEEVerifier(&bind.CallOpts{})
+	if err != nil {
+		return fmt.Errorf("failed to get EspressoTEEVerifier address from BatchAuthenticator contract: %w", err)
+	}
+
+	espressoTEEVerifier, err := bindings.NewEspressoTEEVerifierCaller(verifierAddress, l.L1Client)
+	if err != nil {
+		return fmt.Errorf("failed to create EspressoTEEVerifier contract bindings: %w", err)
+	}
+
+	nitroVerifierAddress, err := espressoTEEVerifier.EspressoNitroTEEVerifier(&bind.CallOpts{})
+	if err != nil {
+		return fmt.Errorf("failed to get EspressoNitroTEEVerifier address from verifier contract: %w", err)
+	}
+
+	nitroVerifier, err := bindings.NewEspressoNitroTEEVerifierCaller(nitroVerifierAddress, l.L1Client)
+	if err != nil {
+		return fmt.Errorf("failed to create EspressoNitroTEEVerifier contract bindings: %w", err)
+	}
+
+	certManagerAddress, err := nitroVerifier.CertManager(&bind.CallOpts{})
+	if err != nil {
+		return fmt.Errorf("failed to get CertManager address from EspressoNitroTEEVerifier contract: %w", err)
+	}
+
+	certManager, err := bindings.NewCertManagerCaller(certManagerAddress, l.L1Client)
+	if err != nil {
+		return fmt.Errorf("failed to create CertManager contract bindings: %w", err)
 	}
 
-	// Decode the attestation off-chain to conserve gas
-	attestationTbs, signature, err := batchAuthenticator.DecodeAttestationTbs(&bind.CallOpts{}, l.Attestation)
+	certManagerAbi, err := bindings.CertManagerMetaData.GetAbi()
 	if err != nil {
-		return fmt.Errorf("failed to decode attestation: %w", err)
+		return fmt.Errorf("failed to create CertManager contract bindings: %w", err)
+	}
+
+	// Verify all CA certiciates in the chain in an individual transaction. This avoids running into block gas limit
+	// that could happen if CertManager verifies the whole certificate chain in one transaction.
+	parentCertHash := crypto.Keccak256Hash(l.Attestation.Document.CABundle[0])
+	for _, cert := range l.Attestation.Document.CABundle {
+		txData, err := createVerifyCertTransaction(certManager, certManagerAbi, cert, true, parentCertHash)
+		if err != nil {
+			return fmt.Errorf("failed to create verify certificate transaction: %w", err)
+		}
+
+		parentCertHash = crypto.Keccak256Hash(cert)
+
+		if txData == nil {
+			// Certificate already verified
```

Comment:
Could you document the function `createVerifyCertTransaction` above and in particular explain why when txData is nil this means the certificate is already verified?